.github/workflows/ci.yaml

```diff
@@ -8,59 +8,55 @@ on:
     branches: [ 'main' ]
 
 jobs:
-  set-matrix:
-    name: Set CI target to matrix
-    runs-on: ubuntu-latest
-    outputs:
-      matrix: ${{ steps.set-matrix.outputs.matrix }}
-
-    steps:
-      - uses: actions/checkout@v3
-      - uses: bazelbuild/setup-bazelisk@v2
-      - run: sudo apt update && sudo apt install -y jq
-      - id: set-matrix
-        run: echo matrix=$(bazel query 'attr("tags", "'amd64'", "//...")' | tr ':' '/' | awk -F '/' '{print $3}' | uniq | jq -Rc . | jq -cs .) >> $GITHUB_OUTPUT
-
-  ci:
-    needs: set-matrix
-    strategy:
-      matrix:
-        target: ${{ fromJson(needs.set-matrix.outputs.matrix) }}
-    name: CI tests for ${{ matrix.target }}
+  ci-images:
+    name: CI tests for images
     runs-on: ubuntu-20.04 # most compatible with debian 11
 
     steps:
-    - uses: actions/checkout@v3
-    - uses: bazelbuild/setup-bazelisk@v2
-    - name: Build and test
-      run: |
-        set -ex
-        for i in $(seq 5); do bazel fetch --loading_phase_threads=1 //${{ matrix.target }}/... && break || sleep 20; done
-        targets=$(bazel query 'attr("tags", "'amd64'", "//${{ matrix.target }}/...")')
-        bazel build --curses=no ${targets}
-        bazel test --curses=no --test_output=errors ${targets}
+      - uses: actions/checkout@v3
+      - name: Mount bazel caches
+        uses: actions/cache@v3
+        with:
+            path: |
+                ~/.cache/bazel
+                ~/.cache/bazel-repo
+            key: bazel-cache-image-${{ hashFiles('**/BUILD.bazel', '**/*.bzl', '**/*.yaml', 'WORKSPACE') }}-${{ github.sha }}
+            restore-keys: |
+                bazel-cache-image-
+                bazel-cache-
+      - name: Free space
+        run: |
+          sudo apt-get remove -y '^ghc-8.*'
```

Comment:
looks like this is causing some failures since the packages isn't there?

Reply:
fixed now.